ZAPD/ZTextureAnimation.cpp

```diff
@@ -0,0 +1,737 @@
+/**
+ * File: ZTextureAnimation.cpp
+ * ZResources defined: ZTextureAnimation, ZTextureAnimationParams (XML declaration not supported for
+ * the latter)
+ * Purpose: extracting texture animating structures from asset files Note: data type is exclusive to
+ * Majora's Mask
+ *
+ * Structure of data:
+ * A texture animation consists of a main array of data of the form
+ *     `SS 00 00 0T PPPPPPPP`,
+ * where `S` is the segment that the functionality is sent to (and read by DLists), `T` is the
+ * "type" (see below), and `P` is a pointer to the type's subsidiary information, which is
+ * invariably just above this main array. Details of this subsidiary data are given below.
+ *
+ * Segment
+ * ===
+ * The segment starts at 1 and is incremented in each entry; the final one is negated, which is used
+ * as the indication to stop reading. The actual segment the game will use to draw is S + 7.
+ *
+ * Types
+ * ===
+ * There are 7 values that `T` can take:
+ *     `0`: A single texture scroll (Implemented by Gfx_TexScroll): subsidiary params are given as
+ *         an array with one entry, a struct `XX YY WW HH` containing xStep, yStep, width, height
+ *         (all u8).
+ *
+ *     `1`: A dual texture scroll (Implementated by Gfx_TwoTexScroll): same as type `0`, but with
+ *         two entries in the params array.
```

Comment:
It would need to be rewritten from a behavioural rather than an extraction perspective, but yes, I assume that MM will eventually have this level of documentation for this, since it's a unique and rather complex set of structs, whether it's written by me or someone a little less prolix.